cmd/upgrade/accountroles/cmd.go

```diff
@@ -376,20 +376,38 @@ func buildCommands(prefix string, accountID string, isUpgradeNeedForAccountRoleP
 	if isUpgradeNeedForOperatorRolePolicies {
 		for credrequest, operator := range aws.CredentialRequests {
 			policyARN := aws.GetOperatorPolicyARN(accountID, prefix, operator.Namespace, operator.Name)
-			policTags := fmt.Sprintf(
-				"Key=%s,Value=%s",
-				tags.OpenShiftVersion, aws.DefaultPolicyVersion,
-			)
-			createPolicy := fmt.Sprintf("aws iam create-policy-version \\\n"+
-				"\t--policy-arn %s \\\n"+
-				"\t--policy-document file://openshift_%s_policy.json \\\n"+
-				"\t--set-as-default",
-				policyARN, credrequest)
-			tagPolicy := fmt.Sprintf("aws iam tag-policy \\\n"+
-				"\t--tags %s \\\n"+
-				"\t--policy-arn %s",
-				policTags, policyARN)
-			commands = append(commands, createPolicy, tagPolicy)
+			_, err := awsClient.IsPolicyExists(policyARN)
+			if err != nil {
+				name := aws.GetPolicyName(prefix, operator.Namespace, operator.Name)
+				iamTags := fmt.Sprintf(
+					"Key=%s,Value=%s Key=%s,Value=%s Key=%s,Value=%s Key=%s,Value=%s",
+					tags.OpenShiftVersion, aws.DefaultPolicyVersion,
+					tags.RolePrefix, prefix,
+					"operator_namespace", operator.Namespace,
+					"operator_name", operator.Name,
+				)
+				createPolicy := fmt.Sprintf("aws iam create-policy \\\n"+
+					"\t--policy-name %s \\\n"+
+					"\t--policy-document file://openshift_%s_policy.json \\\n"+
+					"\t--tags %s",
+					name, credrequest, iamTags)
+				commands = append(commands, createPolicy)
+			} else {
+				policTags := fmt.Sprintf(
+					"Key=%s,Value=%s",
+					tags.OpenShiftVersion, aws.DefaultPolicyVersion,
+				)
+				createPolicy := fmt.Sprintf("aws iam create-policy-version \\\n"+
+					"\t--policy-arn %s \\\n"+
+					"\t--policy-document file://openshift_%s_policy.json \\\n"+
+					"\t--set-as-default",
+					policyARN, credrequest)
+				tagPolicy := fmt.Sprintf("aws iam tag-policy \\\n"+
+					"\t--tags %s \\\n"+
+					"\t--policy-arn %s",
+					policTags, policyARN)
+				commands = append(commands, createPolicy, tagPolicy)
```

Comment:
are you creating the policy again if it already exists?

Reply:
@igoihman If it doesn't exist.    error will not be nil if doesnt exists.  Since this is manual mode we just output the command whatever the error we get . anyways it is user's responsbility to run the aws command